src/annoylib.h

```diff
@@ -782,7 +798,24 @@ template<typename S, typename T, typename Distance, typename Random>
     if (item >= _n_items)
       _n_items = item + 1;
   }
-
+    
+  bool on_disk_build(const char* file) {
+    _on_disk = true;
+    _fd = open(file, O_RDWR | O_CREAT | O_TRUNC, (int) 0600);
+    if (_fd == -1) {
+      _fd = 0;
+      return false;
+    }
+    _nodes_size = 1;
+    ftruncate(_fd, _s * _nodes_size);
+#ifdef MAP_POPULATE
```

Comment:
I think one could test using [madvise](https://man7.org/linux/man-pages/man2/madvise.2.html) with MADV_RANDOM to improve performance: it prevents read-ahead by telling the kernel to expect random access.